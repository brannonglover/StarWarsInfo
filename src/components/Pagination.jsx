import React from 'react';
import strings from '../strings';

const Pagination = ({ page, setPage, latestData }) => {
  return (
    <>
      <button 
        disabled={page === 1} 
        onClick={() => setPage(old => Math.max(old - 1, 1))}
      >{strings.PREVIOUS}</button>
      <span>{page}</span>
      <button 
        disabled={!latestData || !latestData.next} 
        onClick={() => setPage(old => !latestData || !latestData.next ? old : old + 1)}
      >{strings.NEXT}</button>
    </>
  )
}

export default Pagination;