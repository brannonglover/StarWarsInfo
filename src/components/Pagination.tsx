import React from "react";
import strings from "../common/strings";

interface IPagination {
  pageNumber: number;
  setPageNumber: (value: number | ((prevState: number) => number)) => void;
  latestData: any;
}

const Pagination: React.FC<IPagination> = (props: IPagination) => {
  const { pageNumber, setPageNumber, latestData } = props;

  return (
    <>
      <button
        disabled={pageNumber === 1}
        onClick={() => setPageNumber((old: number) => Math.max(old - 1, 1))}
      >
        {strings.PREVIOUS}
      </button>
      <span>{pageNumber}</span>
      <button
        disabled={!latestData || !latestData.next}
        onClick={() =>
          setPageNumber((old) =>
            !latestData || !latestData.next ? old : old + 1
          )
        }
      >
        {strings.NEXT}
      </button>
    </>
  );
};

export default Pagination;
