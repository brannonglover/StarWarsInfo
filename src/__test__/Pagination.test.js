import React from 'react';
import Pagination from '../components/Pagination';
import { cleanup, render } from '@testing-library/react';

afterEach(cleanup);

test('Display buttons for pagination', () => {
  const { asFragment } = render(<Pagination page={1} setPage={() => jest.fn()} />);

  expect(asFragment()).toMatchSnapshot();
});