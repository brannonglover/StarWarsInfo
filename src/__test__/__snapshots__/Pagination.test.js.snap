// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display buttons for pagination 1`] = `
<DocumentFragment>
  <button
    disabled=""
  >
    Previous
  </button>
  <span>
    1
  </span>
  <button
    disabled=""
  >
    Next
  </button>
</DocumentFragment>
`;
